### ISSUE TITLE
Migration from DependencyInjection 7.0 to 9.0 - Error constructing handlers 

### ISSUE THREAD

**Issue by BH4NG** [jbogard/MediatR.Extensions.Microsoft.DependencyInjection#90]
Hi All, 

Been using MediatR since version 7, yesterday I migrated from 8.1 to 9.0. 

Basically I had to inject IPublisher in DbContext, EventDispatcher and CommandHandlers that trigger Events and inject ISender in EventHandlers & Controllers? 

I get following error:
System.InvalidOperationException: Error constructing handler for request of type MediatR.IRequestHandler`2,  Register your handlers with the container.
Cannot resolve 'MediatR.IRequestHandler`2[]' from root provider because it requires scoped service 'xxxRepo'.

the operation continues, the request gets handled, the correct data is found, but my API-call returns 500StatusCode and subsequent calls to the same endpoint are not executed.

Had no issues with unregistered handlers of any kind prior to migrating to 9.0
In my startup I register Handlers first, then the repos, services and finally validators.
Do I need to register IPublisher & ISender as well?

Thanks in advance

**Comment by jbogard** [comment]
Yeah I'm not sure, I've upgraded a few applications without any issues. I would look at the service collection before/after each call in Startup to make sure it has everything and what is overwriting it.

**Comment by rickyzu** [comment]
@jbogard - found our issue and all is working as intended now. The previous developer of the source had not registered one of the IEntityRepository in the DI so the handler that was dependent on it was correctly failing.
